admin/includes/javascript/orders.php

```diff
@@ -0,0 +1,42 @@
+<?php
+// Copy address details to clipboard
+if (isset($action) && $action === 'edit') { ?>
+<script>
+    function copyToClipboard(addressType) {
+        <?php
+        $address_types = ['customer', 'delivery', 'billing'];
+        foreach ($address_types as $address_type) {
+            $addressBlock[$address_type] =
+                $order->$address_type['name'] . "\\r\\n" .
+                (empty($order->$address_type['company']) ? '' : $order->$address_type['company'] . "\\r\\n") .
```

Comment:
Are these double-backslashes necessary? When they're inside double-quotes like this, PHP would already convert `\r` to carriage-return and `\n` as new-line.

If the reason for the double-backslash is to "escape" things for later use with the javascript, then probably using single-quotes in PHP would suffice.